### PR TITLE
Makes flockdrones 24.7% more robust

### PR DESCRIPTION
**Pull request by TobleroneSwordfish**
[BALANCE]
## About the PR 
Gives flockdrones a 25% chance to automatically shoot back when hit (if they are AI controlled and their stunner is not on cooldown)
Makes flockdrones always run away if they are close to a target and their stunner is on cooldown.
Gives flockdrones a 20% chance to shoot at incapacitated targets.

## Why's this needed? 
A staffie with a fire extinguisher can currently tear through quite a large amount of drones with very little chance of being caged, this should make fighting AI drones a little bit more scary and give flockminds/traces more opportunities to capture enemies.

### CODE REVIEW

**Comment by FlameArrow57** [comment]
For first thing, would this be better fixed looking into an ai change making them switch to shoot task faster if possible, or still kind of hard to do?

**Comment by amylizzle** [comment]
> For first thing, would this be better fixed looking into an ai change making them switch to shoot task faster if possible, or still kind of hard to do?

I want to look into this. I was thinking I'd make combat tasks like caging and shooting move the mob into `HEAVYWEIGHT_AI` mode, so response time between actions is 0.2 seconds instead of 1 second.

File: code/mob/living/critter/flock/flockdrone.dm [hunk 1]
```diff
@@ -539,6 +539,12 @@
 	if (!M) return
 	if (isflockmob(M)) return
 	if (!isdead(src) && src.flock)
+		if (!src.controller && prob(25) && (src in oviewers(M)))
```
Comment (amylizzle) [inline code/mob/living/critter/flock/flockdrone.dm:542]:
```suggestion
		if (src.is_npc && prob(25) && (src in oviewers(M)))
```
`is_npc` reflects controller posession as well as dormant state, probably better to use

**Comment by TobleroneSwordfish** [comment]
I like the combat heavyweight idea, though bear in mind it could easily end up with the entire flock in heavyweight (which might actually be fine idk)
This PR is meant to basically hack around the problem of slow AI switching until we can come up with a better solution (maybe just buffing the overall tickrate 🤷‍♀️)

**Comment by FlameArrow57** [comment]
Does heavyweight on shooting fix the first thing, or still needed?

**Comment by TobleroneSwordfish** [comment]
> Does heavyweight on shooting fix the first thing, or still needed?

Well after testing this on local, I fire one shot at a drone and get instantly stunned and caged. I'm going to consider this PR probably unnecessary for now, and also holy shit Amy, heavyweight combat is scary.

**Pull request closed without merging**